content/en/docs/concepts/workloads/controllers/job.md

```diff
@@ -659,6 +674,51 @@ there is no automatic Job restart once the Job status is `type: Failed`.
 That is, the Job termination mechanisms activated with `.spec.activeDeadlineSeconds`
 and `.spec.backoffLimit` result in a permanent Job failure that requires manual intervention to resolve.
 
+### Terminal Job conditions
+
+A Job has two possible terminal states, it ends up either succeeded, or failed,
+and these states are reflected by the presence of the Job conditions `Complete`
+or `Failed`, respectively.
+
+The failure scenarios encompass:
+- the number of pod failures exceeded the specified `.spec.backoffLimit`;
+  see also [Pod backoff failure policy](#pod-backoff-failure-policy)
+- the Job runtime exceeded the specified `.spec.activeDeadlineSeconds`
+- the Job using `.spec.backoffLimitPerIndex` has some indexes failed;
+  see also [Backoff limit per index](#backoff-limit-per-index)
+- the number of failed indexes in the Job has exceeded the specified
+  `spec.maxFailedIndexes`; see also [Backoff limit per index](#backoff-limit-per-index)
+- a failed Pod matches a rule in `.spec.podFailurePolicy`
+   with the `FailJob` action; see also [Pod failure policy](#pod-failure-policy) for more details on how the policy rules can affect evaluation of failures
+
+The success scenarios encompass:
+- the number of succeeded pods reached the specified `.spec.completions`
+- the criteria specified by the `.spec.successPolicy` are met;
+  see also [Success policy](#success-policy)
+
+### Termination of Job pods
+
+Prior to v1.31 the Job terminal conditions are added when the Job termination
+process is triggered, and all Pod finalizers are removed, but some pods may
+still remain running/terminating at that point in time.
+
+Since v1.31, when you enable either the
+`JobManagedBy` or the `JobPodReplacementPolicy` (enabled by default)
+[feature gate](/docs/reference/command-line-tools-reference/feature-gates/), the
+Job controller awaits for termination of all pods before adding a condition
+indicating that the Job is finished (either `Complete` or `Failed`).
+
+The process of terminating all pods may take a substantial amount
+of time, depending on a Pod's `terminationGracePeriodSeconds` (see
+[Pod termination](#docs/concepts/workloads/pods/pod-lifecycle/#pod-termination)),
+and thus adding the terminal Job condition, even if the fate of the Job is
+already determined.
+
+If you want to know the fate of the Job as soon as determined you can use,
+since v1.31, the `FailureTarget` and `SuccessCriteriaMet` conditions, which
+cover all scenarios in which Job controller triggers the Job termination process
+(see [Terminal Job conditions](#terminal-job-conditions)).
```

Comment:
```suggestion
Factors like `terminationGracePeriodSeconds` might increase the amount of time
from the moment that the Job controller adds the `FailureTarget` condition or the
`SuccessCriteriaMet` condition to the moment that all of the Job Pods terminate
and the Job controller adds a [terminal condition](#terminal-job-conditions)
(`Failed` or `Complete`).

You can use the `FailureTarget` or the `SuccessCriteriaMet` condition to evaluate
whether the Job has failed or succeeded without having to wait for the controller
to add a terminal condition.
```

Reply:
I think that this is the most important bit of info. You can also add an example or two to the last paragraph for WHY you'd need this, like 

"For example, you can use this information to quickly decide whether to create a replacement Job, or to make scheduling decisions based on node availability."

Reply:
Rephrased, also added the WHY in a note.